brave/src/main/java/brave/Tracer.java

```diff
@@ -261,6 +263,17 @@ public SpanInScope withSpanInScope(Span span) {
     return currentContext != null ? toSpan(currentContext) : null;
   }
 
+  /**
+   * Conditionally joins a span, or starts a new trace, depending on if a trace context was
+   * extracted from the carrier (usually an incoming request).
+   */
+  public <C> Span nextSpan(Extractor<C> extractor, C carrier) {
```

Comment:
Nice simplification!  I'm on the fence about the name though.  `nextSpan` still feels better since its creating a span from the extracted context.  `extractSpan` feels like its materializing an existing span rather than creating a new one.